test/src/bugs.test.js

```diff
@@ -0,0 +1,116 @@
+    tr.setCurrencyTo("BTC");
+    // 1 BTC to 1 BTC
+    tr.setAmountToProvider(Buffer.from(['0x00', '0x00', '0x00', '0x00', '0x00', '0x00', '0x00', '0x10', '0x10', '0x10', '0x10']));
+    tr.setAmountToWallet(Buffer.from(['0x00', '0x00', '0x00', '0x00', '0x00', '0x05', '0xf5', '0xe1', '0x00']));
+    tr.setDeviceTransactionId(transactionId);
+
+    const payload: Buffer = Buffer.from(tr.serializeBinary());
+    await swap.processTransaction(payload, 840000000000000);
+    const digest: Buffer = Buffer.from(sha256.sha256.array(payload));
+    const signature: Buffer = secp256k1.signatureExport(secp256k1.sign(digest, swapTestPrivateKey).signature);
+    await swap.checkTransactionSignature(signature);
+    const btcAddressParams = getSerializedAddressParametersBTC("84'/0'/0'/1/0", "bech32");
+    await swap.checkPayoutAddress(BTCConfig, BTCConfigSignature, btcAddressParams.addressParameters);
+
+    const checkRequest = swap.checkRefundAddress(BTCConfig, BTCConfigSignature, btcAddressParams.addressParameters);
+
+    // Wait until we are not in the main menu
+    await waitForAppScreen(sim);
+    await sim.navigateAndCompareSnapshots('.', 'nanos_btc_to_btc_overload_swap', [4, 0]);
+    await expect(checkRequest).resolves.toBe(undefined);
+
+    await swap.signCoinTransaction();
+
+    await Zemu.sleep(1000);
+
+    let transport = await sim.getTransport();
+}));
+
+test('[Nano S] Overflow values should be trimmed when funding', zemu("nanos", async (sim) => {
+    const swap = new Exchange(sim.getTransport(), TRANSACTION_TYPES.FUND);
+    const transactionId_base64: string = await swap.startNewTransaction();
+    const transactionId: Buffer = base64url.toBuffer(transactionId_base64);
+    console.log("transactionID %s", transactionId.toString('hex'));
+
+    await swap.setPartnerKey(fundPartnerSerializedNameAndPubKey);
+    await swap.checkPartner(DERSignatureOfFundPartnerNameAndPublicKey);
+
+    var tr = new proto.ledger_swap.NewFundResponse();
+
+    tr.setUserId("John Doe");
+    tr.setAccountName("Card 1234");
+    tr.setInAddress("LKtSt6xfsmJMkPT8YyViAsDeRh7k8UfNjD");
+    tr.setInCurrency("BTC");
+    // 1 BTC
+    tr.setInAmount(Buffer.from(['0x00', '0x00', '0x00', '0x00', '0x00', '0x00', '0x00', '0x05', '0xf5', '0xe1', '0x00']));
+    tr.setDeviceTransactionId(transactionId);
+
+    const payload: Buffer = Buffer.from(tr.serializeBinary());
+
+    const base64_payload: Buffer = Buffer.from(base64url(payload));
+
+    await swap.processTransaction(base64_payload, 10000000);
+
+    const message = Buffer.concat([Buffer.from('.'), base64_payload])
+    const digest: Buffer = Buffer.from(sha256.sha256.array(message));
+    const lol = secp256r1.sign(digest, fundTestPrivateKey).signature;
+
+    process.stdout.write(JSON.stringify(lol) + '\n');
+
+    const signature = secp256r1.signatureExport(lol)
+    //console.log("Length of generated signature %d", sigTemp.length)
+
+    process.stdout.write(JSON.stringify(signature) + '\n');
```

Comment:
remove debug prints, and rename `lol` to something :P

Reply:
lol

Reply:
My patience with JS illustrated right there.